### PR TITLE
Added more PhpUnit tests

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

- added some tests

### Comments

To exclude slow running tests;

```bash
phpunit --exclude Base,runInSeparateProcess
```

### CODE REVIEW

**Review by ADDISON74 — APPROVED** [review]

File: app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Websites.php [hunk 1]
```diff
@@ -42,7 +42,7 @@ public function getProduct()
     /**
      * Get store ID of current product
      *
-     * @return int
+     * @return string
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Websites.php:45]: https://github.com/OpenMage/magento-lts/blob/df37dba44808847ccbb90a4ec8a53cb6065e832c/app/code/core/Mage/Catalog/Model/Product.php#L215-L215
```suggestion
     * @return int|string
```

Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Websites.php:45]: I think i have xdebuged it, and it is always a string.
 
 The method annotation should be changed. (?)

Comment (kiatng) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Websites.php:45]: Consider 
```php
$product = Mage::getModel('catalog/product')
       ->setStoreId($storeId)
       ->load($id);
``` 
OpenMage is not type strict: `$storeId` and `$id` can be type string or int. However, the column `store_id` is type `smallint(5)` in the EAV tables, so `int` is more correct in docblock. Therefore, I suggest `int` first then `string`. We can leave the annotation as is.

Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Websites.php:45]: > However, the column store_id is type smallint(5) in the EAV tables, so int is more correct in docblock.

Thats true. Mhhh ... if we want to reflect the DB column type we should turn it into "int" - everywhere.

Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Websites.php]: Reverted some changes and cast to int (in a few places).

**Review by ADDISON74 — DISMISSED** [review]

**Review by ADDISON74 — DISMISSED** [review]

File: app/code/core/Mage/Eav/Model/Entity/Attribute/Backend/Abstract.php [hunk 2]
```diff
@@ -222,7 +222,7 @@ public function getDefaultValue()
      *
      * @param Varien_Object $object
      * @throws Mage_Eav_Exception
-     * @return bool
+     * @return $this|bool
```
Comment (kiatng) [inline app/code/core/Mage/Eav/Model/Entity/Attribute/Backend/Abstract.php:225]: I cannot see where it is returning `$this`. Can you point it out?

Comment (sreichel) [reply app/code/core/Mage/Eav/Model/Entity/Attribute/Backend/Abstract.php:225]: It is to fix this:

```
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   app/code/core/Mage/Weee/Model/Attribute/Backend/Weee/Tax.php                                                                                                                             
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  41     Return type ($this(Mage_Weee_Model_Attribute_Backend_Weee_Tax)) of method Mage_Weee_Model_Attribute_Backend_Weee_Tax::validate() should be compatible with return type (bool) of method  
         Mage_Eav_Model_Entity_Attribute_Backend_Abstract::validate()                                                                                                                             
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

```

Comment (kiatng) [reply app/code/core/Mage/Eav/Model/Entity/Attribute/Backend/Abstract.php:225]: Nasty.

Comment (sreichel) [reply app/code/core/Mage/Eav/Model/Entity/Attribute/Backend/Abstract.php:225]: Yep. I dont like it, but i dont have a better idea?

Docs only reflect what we have to deal with ...

File: app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Websites.php (hunk 1 above)
Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Websites.php]: @kiatng any objections to cast store id to int?

Comment (kiatng) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Websites.php]: No objection to cast store ID to int.

File: app/code/core/Mage/Catalog/Model/Category.php [hunk 3]
```diff
@@ -314,7 +314,7 @@ public function move($parentId, $afterCategoryId)
     /**
      * Retrieve default attribute set id
      *
-     * @return int
+     * @return string
```
Comment (kiatng) [inline app/code/core/Mage/Catalog/Model/Category.php:317]: Attribute set ID is stored as int in the DB. Is it more correct to return as int? I'm aware that reading int from DB converts to string. It's problematic. But in the spirit of strict typing, it should be int. Your thought?

Comment (sreichel) [reply app/code/core/Mage/Catalog/Model/Category.php]: Yes. Same as for "store_id". Cast to int?

**Review by kiatng — DISMISSED** [review]

File: app/code/core/Mage/Catalog/Model/Product.php [hunk 4]
```diff
@@ -217,7 +217,7 @@
  * @method $this setStoreIds(array $storeIds)
  * @method array getSwatchPrices()
  *
- * @method int getTaxClassId()
+ * @method string getTaxClassId()
```
Comment (kiatng) [inline app/code/core/Mage/Catalog/Model/Product.php:220]: int?

**Review by ADDISON74 — APPROVED** [review]

**Review by kiatng — APPROVED** [review]

**Comment by fballiano** [comment]
this PR should be about unit tests and it's mainly about phpstan, but also, did you notice it introduces 3-4 new errors in phpstan?

**Comment by sreichel** [comment]
to write the tests, i have to view the files. when phpstan issues can be fixed at the same time ....

yes, i did, fixing one phpstan error can reveal others, but this can become endless story to fix all follow-up errors. Better +5 and -150 for now? They will be fixed later.

**Pull request merged**